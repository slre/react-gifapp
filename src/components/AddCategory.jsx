import { useState } from "react"

const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('');
    const onInputChange = (event)=>{
        setInputValue(event.target.value )
    }
    const onSubmit = (event)=>{
        event.preventDefault();
        if(inputValue.trim().length <=1) return;
        setCategories(categories => [inputValue,...categories]);
        setInputValue('');
    }    
    return (
        <form onSubmit={(event)=>onSubmit(event)}>
            <input 
            type="text" 
            placeholder="Search for a gif"
            value={inputValue}
            onChange={ (event)=>onInputChange(event)}
            />
            </form>
    )
        
        
}

export default AddCategory