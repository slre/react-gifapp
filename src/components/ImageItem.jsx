import React from 'react'
import PropTypes from 'prop-types';

export const ImageItem = ( {id,title,url} ) => {
    //console.log(url);
  return (
    <>
    <div className='card'>
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
    </>
    
  )
}
ImageItem.propTypes = {
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired

}