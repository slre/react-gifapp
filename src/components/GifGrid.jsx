import React, { useEffect, useState } from 'react'
import { ImageItem } from './ImageItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid  = ({category}) => {
  
   
    
    const { images, isLoading  } = useFetchGifs(category);
    return (
        <>
            <h1>{category}</h1>
            {
                // isLoading === true
                // ? ( <h2>Loading...</h2> )
                // : null

                isLoading && ( <h2>Loading...</h2> )
            }
            <h2></h2>
            <div className='card-grid'>
                {images.map((img) => (
                <ImageItem 
                    key={img.id} 
                    //title={img.title} 
                    //url={img.url.url}
                    { ...img}  
                />
            ))}
            </div>
      
        </>
  );
}

export default GifGrid