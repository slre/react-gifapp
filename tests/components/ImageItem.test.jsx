import { act } from "react";
import { ImageItem } from "../../src/components/ImageItem";
import { render, screen } from '@testing-library/react';



describe('Testing <ImageItem/>', () => { 
    const title="Naruto";
    const url = "http://api.com/";
    test('debe hacer match', () => { 
       
        const { container } = render( <ImageItem title={title} url={url} /> );
        expect(container).toMatchSnapshot();
        //render(<ImageItem title='Some title' url='SomeURL'/>)

    }); 

    test('Should print img url and alt ', () => { 
        render(<ImageItem title={title} url={url}/>);
        //screen.debug();
    
        //expect(screen.getByRole('img').src).toBe(url);

        const { src,alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);

     })

     test('should show title in component', () => { 
       // act(()=>{
            render(<ImageItem title={title} url={url}/>);
       // })
        
        expect( screen.getByText(title)).toBeTruthy();
        
      })
});