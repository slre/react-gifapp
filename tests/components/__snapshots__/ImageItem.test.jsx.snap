// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ImageItem/> debe hacer match 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="http://api.com/"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
